### PR TITLE
[ENH,WIP] Add OpenAPI Specification

### LINKED ISSUE #21: Create OpenAPI specification manually
Using stoplight.io ([our workspace](https://psychoinformaticslab.stoplight.io/)) to create a manual specification of API endpoints.

### PR DESCRIPTION
**Pull request by jdkent**
closes #21
Here is a draft of the specification adding the dataset and study routes.

### CODE REVIEW

File: neurostuff/openapi/neurostuff-api.v1.yml [hunk 1]
```diff
@@ -0,0 +1,726 @@
...
+      summary: Your GET endpoint
+      tags: []
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/condition'
+      description: I do not rightly know (?)
```
Comment (jdkent) [inline neurostuff/openapi/neurostuff-api.v1.yml:202]: I do not really know what condition is supposed to represent

File: neurostuff/openapi/neurostuff-api.v1.yml [hunk 2]
```diff
@@ -0,0 +1,726 @@
...
+        - analysis
+        - created_at
+        - doi
+        - name
+        - '@context'
+    analysis:
+      title: analysis
+      type: object
+      properties:
+        '@context':
```
Comment (jdkent) [inline neurostuff/openapi/neurostuff-api.v1.yml:420]: How should I describe all these keys (with `@` symbols), I'm assuming this is a part of the JSON-LD structure.

File: neurostuff/openapi/neurostuff-api.v1.yml [hunk 3]
```diff
@@ -0,0 +1,726 @@
...
+              example: 113751
+              description: neurovault ID of the image
+            image_type:
+              type: string
+              example: statistic_map
+              description: the type of data the image represents
+            is_thresholded:
+              type: boolean
+              description: whether the image is thresholded
+            is_valid:
```
Comment (jdkent) [inline neurostuff/openapi/neurostuff-api.v1.yml:592]: What does it mean to be a valid image?

File: neurostuff/openapi/neurostuff-api.v1.yml [hunk 4]
```diff
@@ -0,0 +1,726 @@
...
+              example: '"2019-01-29T03:19:10.274381Z"'
+              description: time the image was modified
+            name:
+              type: string
+              example: Positive RPEs
+              description: name of the image
+            not_mni:
+              type: boolean
+              description: whether the image is not in MNI space
+            perc_bad_voxels:
```
Comment (jdkent) [inline neurostuff/openapi/neurostuff-api.v1.yml:615]: what makes a voxel bad?

File: neurostuff/openapi/neurostuff-api.v1.yml [hunk 5]
```diff
@@ -0,0 +1,726 @@
...
+              type: number
+              format: float
+              example: 75.5340233916703
+              minimum: 0
+              maximum: 100
+              description: percentage of bad voxels (what makes voxels bad?)
+            perc_voxels_outside:
```
Comment (jdkent) [inline neurostuff/openapi/neurostuff-api.v1.yml:622]: what are the voxels outside of? (the mask?)

File: neurostuff/requirements.txt [hunk 6]
```diff
@@ -1,57 +1,20 @@
-aniso8601==8.0.0
-Babel==2.8.0
-blinker==1.4
-cachetools==4.1.1
-certifi==2020.6.20
-chardet==3.0.4
-click==7.1.2
-dnspython==2.0.0
-email-validator==1.1.1
+wheel
```
Comment (tyarkoni) [inline neurostuff/requirements.txt:1]: I'm on board with not pinning to specific versions, but we should probably require minimum versions, as using old versions of some of these packages will definitely cause problems.

**Review by tyarkoni — COMMENTED** [review]
Looks great, thanks! Only one minor comment about versions in requirements.txt, which you can take or leave as you please.

**Comment by tyarkoni** [comment]
Okay to merge this, @jdkent? This is all still a WIP, so it's not a big deal if some things are broken—there will be lots of changes throughout the codebase. But I'd like to try to deploy a prototype to our server so we can start developing against it.

**Comment by jdkent** [comment]
yes, let me resolve the conflict.

File: neurostuff/tests/api/test_openapi.py [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import schemathesis
+from hypothesis import settings
+import pytest
+
+#from ...core import app
+
+
+# schema = schemathesis.from_wsgi("/api/openapi.json", app)
+
+# @pytest.mark.skip(reason="Currently cannot get these tests to work")
```
Comment (Stranger6667) [inline neurostuff/tests/api/test_openapi.py:10]: Hi! My name is Dmitry, and I am the author of Schemathesis :) Just letting you know that if you need any help with integrating it, I'd be happy to help :)

Comment (jdkent) [reply neurostuff/tests/api/test_openapi.py:10]: Thank you! I'll open a separate pull request to get your help on using schemathesis

**Pull request merged**